src/server/active_tran_server.cpp

```diff
@@ -48,6 +48,34 @@ active_tran_server::~active_tran_server ()
     }
 }
 
+void active_tran_server::parse_server_host (std::string host, const char *db_name, bool *connected)
+{
+  std::string m_ps_hostname;
+  auto col_pos = host.find (":");
+  long port = -1;
+
```

Comment:
Shouldn't col_pos be checked?

```suggestion
  if (col_pos < 1 || col_pos >= host.length () - 1)
    {
      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_HOST_PORT_PARAMETER, 2, prm_get_name (PRM_ID_PAGE_SERVER_HOSTS),
	      host.c_str ());
      return ER_HOST_PORT_PARAMETER;
    }
```